Assets/Scripts/InventoryMenuItemToggle.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class InventoryMenuItemToggle : MonoBehaviour
+{
+    [Tooltip("The image component used to show the associated object's icon.")]
+    [SerializeField]
+    private Image iconImage;
+
+    public static event Action<InventoryObject> InventoryMenuItemSelected;    
+
+    private InventoryObject associatedInventoryObject;
+
+    public InventoryObject AssociatedInventoryObject
+    {
+        get { return associatedInventoryObject; }
+        set
+        {
+            associatedInventoryObject = value;
+            iconImage.sprite = associatedInventoryObject.Icon;
+        }
+    }
+    /// <summary>
```

Comment:
I would actually recommend adding a line here, but I think that's more my personal preference than Style Conventions?